app/components/govuk_component/header_component.rb

```diff
@@ -77,7 +78,12 @@ def link?
 
     def call
       if link?
-        link_to(text, href, class: "govuk-header__link")
+        case method
+        when :get
+          link_to(text, href, class: "govuk-header__link")
+        when :post, :patch, :put, :delete
+          link_to(text, href, method: method, class: "govuk-header__link")
+        end
```

Comment:
Should we use `button_to` here instead?

I'm not sure how that would affect styling though.

Reply:
Yes, that would affect the styling as we have links on navbar

Reply:
My concern is that links with other actions won't progressively enhance. I wonder if it's possible to style the button as a navigation item link.